### ISSUE TITLE
Wrong behavior when choosing jpeg optimizer

### ISSUE THREAD

**Issue by n2j7** [psliwa/image-optimizer#23]
When you have installed only `jpegoptim` on your system there are no ability to optimize your images due to wrong choose algorithm.
In documentation you said: "first of two optimizations will be executed: jpegtran or jpegoptim", but in the code:

```
$this->optimizers['jpeg'] = $this->optimizers['jpg'] = new ChainOptimizer(array(
            $this->unwrap($this->optimizers['jpegtran']),
            $this->unwrap($this->optimizers['jpegoptim']),
        ), true);
```

There are new instance of `ChainOptimizer` with `$executeFirst=true` and when it optimize (with my comments)

```
public function optimize($filepath)
    {
        foreach($this->optimizers as $optimizer) { // get first optimizer (jpegtran) which wasn't installed
            $optimizer->optimize($filepath); // try to optimize image with it

            if($this->executeFirst) break; // break due to $this->executeFirst = true
        }
    }
```

And we have no chance to change the order of optimizers.
I can send a pull request for you, but which decision i have to choose: allow user to set up optimizers by hands or check optimizer availability in OS

**Comment by nerijuso** [comment]
Same problem for me. If I have only jpegoptim, your library still trying to optimize with jpegtran and process stops. Image can not be optimized.

**Comment by matjazpotocnik** [comment]
@n2j7 , how did you solve this? Did you go for option b, check optimizer availability in OS?

**Comment by n2j7** [comment]
I can solve this problem in any way, but I have asked for advice from author with no comments yet.
You have to install `jpegtran`  to make it worked for you without any patches.

**Comment by matjazpotocnik** [comment]
I would be interested for the patch :-) My idea is: if optimizer is not found (not installed), even if you supply the path to the optimizer, then simply skip to the next available optimizer.

**Comment by matjazpotocnik** [comment]
@n2j7 , are you interested in making PR? Even if it's not accepted from the author, it would be of great help for me. Thank you.

**Comment by n2j7** [comment]
Maybe this code will help you. As is from project.
```
$tg = new ImageOptimizer\TypeGuesser\SmartTypeGuesser();
$detected_type = $tg->guess(EVO_S3_DEST_DIR . $dest_name);

$factory = new \ImageOptimizer\OptimizerFactory(array(
	'ignore_errors' => true,
	'jpegoptim_options' => array('--strip-all', '-m85', '--all-normal')
));

// this workaround due to bug https://github.com/psliwa/image-optimizer/issues/23
if ($detected_type == ImageOptimizer\TypeGuesser\TypeGuesser::TYPE_JPEG) {
	$optimizer = $factory->get('jpegoptim');
}
else {
	$optimizer = $factory->get();
}

$optimizer->optimize(EVO_S3_DEST_DIR . $dest_name);
```
I'll create a patch if i have a time.

**Comment by matjazpotocnik** [comment]
@n2j7 Thank you!

**Comment by psliwa** [comment]
I think it has been fixed: https://github.com/psliwa/image-optimizer/pull/26

Fix is in `1.1.0` version.

**Comment by n2j7** [comment]
Cool.Thank you! I have missed your commits but `1.1.0` works fine.
